lsp/server/src/validatorManager.ts

```diff
@@ -0,0 +1,162 @@
+import { Diagnostic } from 'vscode-languageserver';
+
+import { JSValidator } from './validator/jsValidator';
+import { HTMLValidator } from './validator/htmlValidator';
+import { GraphQLValidator } from './validator/gqlValidator';
+import { DiagnosticSection } from './validator/baseValidator';
+
+import { TextDocument } from 'vscode-languageserver-textdocument';
+
+import { DiagnosticSettings } from './diagnostic/DiagnosticSettings';
+import { OversizedRecord } from './diagnostic/gql/over-sized-record';
+import { AdaptersLocalChangeNotAware } from './diagnostic/js/adapters-local-change-not-aware';
+import { MobileOfflineFriendly } from './diagnostic/html/mobileOfflineFriendly';
+
+import type { BaseValidator, SupportedType } from './validator/baseValidator';
+
+/**
+ * The ValidatorManager class manages a collection of BaseValidator instances and coordinates the validation process for documents.
+ * It filters relevant validators based on the document's language, applies each to designated sections of the document, and aggregates diagnostics
+ * for consistent, language-specific validation. This class centralizes validation logic, enabling efficient, extensible diagnostic processing
+ * across multiple validators.
+ */
+export class ValidatorManager {
+    // Store all available validators
+    private validators: BaseValidator<SupportedType>[] = [];
+
+    private constructor() {}
+
+    /**
+     * Adds a validator to the manager’s collection.
+     * @param validator The validator to be added.
+     */
+    public addValidator(validator: BaseValidator<SupportedType>) {
+        this.validators.push(validator);
+    }
+
+    /**
+     * Validate a document by applying all relevant validators based on language.
+     * @param setting The diagnostic settings.
+     * @param document The document to validate.
+     * @param extensionName The name of the extension (sets diagnostic source).
+     * @returns A promise resolving to an array of diagnostics.
+     */
+    async validateDocument(
+        setting: DiagnosticSettings,
+        document: TextDocument,
+        extensionName: string
+    ): Promise<Diagnostic[]> {
+        const qualifiers = this.validators.filter(
```

Comment:
nitpick:  something like below ```suggestion
        const validatorsNeedToRun = this.validators.filter(
```